### ISSUE TITLE
MAP on VOC07 test

### ISSUE THREAD

**Issue by panda9095** [AlexeyAB/darknet#696]
Hi,

I follow the training steps in ReadMe to train the VOC model. 
But the MAP on VOC07 test is only 71.8. It seems quite farm from yolov2's performance which is about 76.8.
Does anyone also try the training and what is your MAP? And any advice I can follow to improve the MAP?

Thx.

**Comment by AlexeyAB** [comment]
Hi,
* Did you use yolo-voc.cfg or tiny-yolo-voc.cfg?
* What `batch, subdivision, width, height and random` did you use for training and detection?
* To get the maximum mAP you can try to train multiple times and get the best weights-file.

**Comment by panda9095** [comment]
Thx for the reply.

> Did you use yolo-voc.cfg or tiny-yolo-voc.cfg?

Yes, I used yolo-voc.cfg.

> What batch, subdivision, width, height and random did you use for training and detection?

bacth=64, subdivision=32. Other options remain unchanged.

> To get the maximum mAP you can try to train multiple times and get the best weights-file.

By "train multiple times", do you mean start multiple training process independently but with different learning rate schedules? Or just with the same schedules?

**Comment by AlexeyAB** [comment]
Start multiple training process independently with the same learning rate schedules. Every time, the accuracy changes a little.
And try to use subdivision=8 if your GPU allow this.

Did you train 80200 iterations?

**Comment by panda9095** [comment]
Train 50200 iterations by default.  @AlexeyAB 

* I saw you suggested 100200 iterations in another issue, and I am wondering another 50000 iterations training will boost the MAP that much. But I will try continuing training to reach 100000 iterations. 

* Why would the subdivision affect the performance?

Thx.

**Comment by AlexeyAB** [comment]
> Why would the subdivision affect the performance?

`real_batch_size = cfg-batch / cfg-subdivision`: https://github.com/AlexeyAB/darknet/blob/15139eb6abd3c6c05a4046b5ac917de42c678827/src/parser.c#L608

So during training we iterate in the loop ` from 0 to subdivision` and do forward-backward-update for `real_batch_size`:  
* `n` - subdivision
* `batch` - real_batch_size

https://github.com/AlexeyAB/darknet/blob/15139eb6abd3c6c05a4046b5ac917de42c678827/src/network.c#L309-L320


So:
* if you use batch=64 subdivision=32 then `real_batch_size = 2` and will be loaded 64 images
* if you use batch=64 subdivision=8 then `real_batch_size = 8` and will be loaded 64 images

The bigger the real_batch_size - the better network can be generalizied.
